File: 软件设计/模式/设计模式解析/Chapter 10/Example 10-1/Rectangle.cpp
#include "Rectangle.h"

Rectangle::Rectangle(double x1, double y1, double x2, double y2) 
{
    _x1= x1; _y1= y1; _x2= x2; _y2= y2;
}

Rectangle::~Rectangle(void)
{
}

void Rectangle::draw()
{
    drawLine( _x1, _y1, _x2, _y1);
    drawLine( _x2, _y1, _x2, _y2);
    drawLine( _x2, _y2, _x1, _y2);
    drawLine( _x1, _y2, _x1, _y1);
}

File: 其他语言/C#/C# 与 NET4 高级程序设计/Chapter 9/MIInterfaceHierarchy/Rectangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace MIInterfaceHierarchy
{
    class Rectangle : IShape
    {
        public int GetNumberOfSides()
        { return 4; }

        public void Draw()
        { Console.WriteLine("Drawing..."); }

        public void Print()
        { Console.WriteLine("Prining..."); }
    }
}
